src/Owin.Security.Providers.Discord/DiscordAuthenticationHandler.cs

```diff
@@ -0,0 +1,252 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.Owin.Infrastructure;
+using Microsoft.Owin.Logging;
+using Microsoft.Owin.Security;
+using Microsoft.Owin.Security.Infrastructure;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Owin.Security.Providers.Discord.Provider;
+
+namespace Owin.Security.Providers.Discord
+{
+    public class DiscordAuthenticationHandler : AuthenticationHandler<DiscordAuthenticationOptions>
+    {
+        private const string XmlSchemaString = "http://www.w3.org/2001/XMLSchema#string";
+        private const string TokenEndpoint = "https://discordapp.com/api/oauth2/token";
+        private const string UserInfoEndpoint = "https://discordapp.com/api/users/@me";
+
+        private readonly ILogger _logger;
+        private readonly HttpClient _httpClient;
+
+        public DiscordAuthenticationHandler(HttpClient httpClient, ILogger logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        protected override async Task<AuthenticationTicket> AuthenticateCoreAsync()
+        {
+            AuthenticationProperties properties = null;
+
+#if !DEBUG
```

Comment:
can you remove the compiler directive? if you are looking to catch the error while debugging there are numerous things you can invoke System.Diagnostics.Debugger.Break() inside of an if System.Diagnostics.Debugger.IsAttached  == true statement